src/app/Shared/Components/DurationUnitSelect.tsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { MenuToggle, MenuToggleElement, Select, SelectList, SelectOption } from '@patternfly/react-core';
+import { TFunction } from 'i18next';
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+
+export enum DurationUnit {
+  HOUR,
+  MINUTE,
+  SECOND,
+}
+
+const SupportedDurationUnits = Object.values(DurationUnit).filter((entry) => typeof entry === 'string');
+
+export const convertDurationToSeconds = (duration: number, unit: DurationUnit): number => {
```

Comment:
This looks like it would be a good candidate to make into a method of the Duration type, or else to at least accept `Duration` as a parameter.

```typescript
export class Duration {
  constructor(readonly value: number, readonly unit: DurationUnit) { }

  convertTo(unit: DurationUnit): Duration {
    // convert between this.unit and unit parameter, return this.duration * scalar factor
  }
}

// elsewhere...

const dur = new Duration(10, DurationUnit.MINUTE);
const seconds = dur.convertTo(DurationUnit.SECOND).duration;
```

Reply:
> This looks like it would be a good candidate to make into a method of the Duration type, or else to at least accept Duration as a parameter.

Oh okayy right, I was writing with Golang styles of pointer receiver haha. Updated now. I still want to keep the interface type though (just for easier serialization of Redux state, i.e. avoid additional sanitizing).